File: java/dwxquickfix/ExecutionReport.java
package dwxquickfix;

import java.util.Date;
import java.util.TimeZone;
import java.text.SimpleDateFormat;

public class ExecutionReport {
    
    int ClOrdID;
    String Symbol;
    int Side;
    double Price;
    int OrdType;
    String OrdStatus;
    int OrderQty;
    int MinQty;
    int CumQty;
    int LeavesQty;
	Date transactTime;
	
	SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss.SSS");
    
    public ExecutionReport(int ClOrdID, String Symbol, int Side, double Price, int OrdType, String OrdStatus, int OrderQty, int MinQty, int CumQty, int LeavesQty, Date transactTime) {
        this.ClOrdID = ClOrdID;
        this.Symbol = Symbol;
        this.Side = Side;
        this.Price = Price;
        this.OrdType = OrdType;
        this.OrdStatus = OrdStatus;
        this.OrderQty = OrderQty;
        this.MinQty = MinQty;
        this.CumQty = CumQty;
        this.LeavesQty = LeavesQty;
		this.transactTime = transactTime;
		
		sdf.setTimeZone(TimeZone.getTimeZone("UTC"));
    }
	
	public String toString() {
		return "\nExecution report: "
			 + "\ntransactTime: " + sdf.format(transactTime)
		     + "\nClOrdID: " + ClOrdID
			 + "\nOrdStatus: " + OrdStatus
			 + "\nSide: " + Side
			 + "\nPrice: " + Price
			 + "\nOrdType: " + OrdType
			 + "\nOrderQty: " + OrderQty
			 + "\nMinQty: " + MinQty
			 + "\nCumQty: " + CumQty
			 + "\nLeavesQty: " + LeavesQty;
	}
	
	// void toString() {
        // return "ClOrdID: "  ClOrdID + " | Symbol: " + Symbol + " | Side: " + Side + " | Price: " + Price, 
                 // + " | OrdType: " + OrdType + " | OrdStatus: " + OrdStatus + " | OrderQty: " + OrderQty, 
                 // + " | MinQty: " + MinQty + " | CumQty: " + CumQty + " | LeavesQty: " + LeavesQty;
	// }
}

File: python/dwxquickfix/execution_report.py
# -*- coding: utf-8 -*-
"""
    execution_report.py
    @author: Darwinex Labs (www.darwinex.com), 2021-*
    
    execution_report - A data structure to hold execution reports
"""

class execution_report():
    
    # Side: 1=buy, 2=sell
    def __init__(self, ClOrdID, Symbol, Side, Price, OrdType, 
                 OrdStatus, OrderQty, MinQty, CumQty, LeavesQty):

        self.ClOrdID = ClOrdID
        self.Symbol = Symbol
        self.Side = Side
        self.Price = Price
        self.OrdType = OrdType
        self.OrdStatus = OrdStatus
        self.OrderQty = OrderQty
        self.MinQty = MinQty
        self.CumQty = CumQty
        self.LeavesQty = LeavesQty

    def __str__(self):
        return (f'ClOrdID: {self.ClOrdID}, symbol: {self.Symbol}, Side: {self.Side}, Price: {self.Price}, '
                f'OrdType: {self.OrdType}, OrdStatus: {self.OrdStatus}, OrderQty: {self.OrderQty}, '
                f'MinQty: {self.MinQty}, CumQty: {self.CumQty}, LeavesQty: {self.LeavesQty}')
